ghost/core/core/frontend/web/middleware/frontend-caching.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file Middleware to set the appropriate cache headers on the frontend
+ */
+const config = require('../../../shared/config');
+const shared = require('../../../server/web/shared');
+const {api} = require('../../services/proxy');
+
+async function getFreeTier() {
+    const controller = api.tiers;
+    let response = await controller.browse();
+    let freeTier = response.tiers.find(tier => tier.type === 'free');
+    return freeTier;
+}
+
+/**
+ * 
+ * @param {object} member 
+ * @returns string|null - The member's active tier, or null if the member has more than one active subscription
+ */
+function calculateMemberTier(member, freeTier) {
+    const activeSubscriptions = member.subscriptions.filter(sub => sub.status === 'active');
+    let memberTier;
+    if (activeSubscriptions.length === 0) {
+        memberTier = freeTier; // how do we get the free tier's ID in this context? 
+    } else if (activeSubscriptions.length === 1) {
+        memberTier = activeSubscriptions[0].tier;
+    } else {
+        // Member has more than one active subscription
+        // This is rare, but it can happen. 
+        // In this case, don't bother caching the content
+        return null;
+    }
+    return memberTier;
+}
+
+const getMiddleware = (freeTier) => {
```

Comment:
Feels like there should be a better way to do this 🤔

If the member has an active subscription, we can get their tier (and its object ID) from the `req.member` or `res.locals.member` object. 

But if the member is a 'free' member, there isn't any product or subscription to grab the tier from, so we just fallback to 'free'. Somehow we need to get the freeTier into the context of the middleware. Trying to find a way to inject this into the middleware so we can test it more easily — this is the best I've come up with so far but it feels messy.

Reply:
Found a somewhat cleaner way to do this, but I'm still not super satisfied with it. Will put more thought into it, but resolving for now.